plugin-azure-server/src/main/kotlin/jetbrains/buildServer/clouds/azure/arm/KubePodNameGenerator.kt

```diff
@@ -0,0 +1,134 @@
+package jetbrains.buildServer.clouds.azure.arm
+
+import jetbrains.buildServer.clouds.CloudException
+import jetbrains.buildServer.log.Loggers
+import jetbrains.buildServer.serverSide.BuildServerAdapter
+import jetbrains.buildServer.serverSide.BuildServerListener
+import jetbrains.buildServer.serverSide.ServerPaths
+import jetbrains.buildServer.serverSide.executors.ExecutorServices
+import jetbrains.buildServer.util.EventDispatcher
+import jetbrains.buildServer.util.FileUtil
+import jetbrains.buildServer.util.StringUtil
+import org.jetbrains.annotations.NotNull
+import java.io.File
+import java.io.IOException
+import java.nio.charset.Charset
+import java.util.*
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.locks.ReadWriteLock
+import java.util.concurrent.locks.ReentrantReadWriteLock
+
+
+class KubePodNameGenerator(@NotNull eventDispatcher: EventDispatcher<BuildServerListener>,
+                           @NotNull  executorServices: ExecutorServices) {
+
+    private val myCounters: ConcurrentMap<String, AtomicInteger> = ConcurrentHashMap()
+    private val myIndexTouchedMaps: ConcurrentMap<String, AtomicBoolean> = ConcurrentHashMap()
+    private var myIndexStorage: File = File(ServerPaths.PLUGIN_DATA_DIRECTORY_NAME, "azureKubeIndex")
+    private val myIsAvailable: AtomicBoolean = AtomicBoolean(false)
+    private val myLock: ReadWriteLock = ReentrantReadWriteLock()
+
+    init{
+        if (!myIndexStorage.exists()) {
+            myIndexStorage.mkdirs()
+        }
+        if (!myIndexStorage.isDirectory) {
+            throw CloudException("Unable to create a directory for azure arm plugin to monitor VM indexes")
+        }
+        myIsAvailable.set(true)
+
+        loadIndexes(myIndexStorage)
+
+        eventDispatcher.addListener(object : BuildServerAdapter() {
+            override fun serverShutdown() {
+                myIsAvailable.set(false)
+                storeIndexes()
+            }
+        })
+        executorServices.normalExecutorService.scheduleWithFixedDelay(this::storeIndexes, 60, 60, TimeUnit.SECONDS)
+    }
+
+    @Synchronized
+    private fun loadIndexes(@NotNull indexStorage: File) {
+        val indexes = indexStorage.listFiles() ?: return
+        for (indexFile in indexes) {
+            if (!indexFile.name.endsWith(".idx")) continue
+            val indexName = indexFile.name.substring(0, indexFile.name.length - 4)
+            try {
+                val indexValueFromFile = FileUtil.readText(indexFile).toInt()
+                if (indexValueFromFile > 0) {
+                    myCounters.putIfAbsent(indexName, AtomicInteger(0))
```

Comment:
If you your keen on less lines you can use `computeIfAbsent` to turn each place `putIfAbsent` is used into a one liner.
```
myCounters.computeIfAbsent(indexName, () -> AtomicInteger(0)).set(indexValueFromFile);
```

Reply:
Your suggestion is not syntactically correct. But I like the idea behind the suggestion. Will update to reflect below: 
```
myCounters.computeIfAbsent(indexName) { AtomicInteger(indexValueFromFile) }
```